src/probnum/diffeq/odefiltsmooth/kalman_odesolution.py

```diff
@@ -0,0 +1,120 @@
+"""ODE solutions returned by Gaussian ODE filtering."""
+
+import typing
+
+import numpy as np
+
+import probnum._randomvariablelist as pnrv_list
+import probnum.filtsmooth as pnfs
+import probnum.random_variables as pnrv
+import probnum.type
+import probnum.utils
+
+from ..odesolution import ODESolution
+
+
+class KalmanODESolution(ODESolution):
+    """Gaussian IVP filtering solution of an ODE problem.
+
+    Recall that in ProbNum, Gaussian filtering and smoothing is generally named "Kalman".
+
+    Parameters
+    ----------
+    kalman_posterior
+        Gauss-Markov posterior over the ODE solver state space model.
+        Therefore, it assumes that the dynamics model is an :class:`Integrator`.
+
+    See Also
+    --------
+    GaussianIVPFilter : ODE solver that behaves like a Gaussian filter.
+    KalmanPosterior : Posterior over states after Gaussian filtering/smoothing.
+
+    Examples
+    --------
+    >>> from probnum.diffeq import logistic, probsolve_ivp
+    >>> from probnum import random_variables as rvs
+    >>> initrv = rvs.Constant(0.15)
+    >>> ivp = logistic(timespan=[0., 1.5], initrv=initrv, params=(4, 1))
+    >>> solution = probsolve_ivp(ivp, method="ekf0", step=0.1)
+    >>> # Mean of the discrete-time solution
+    >>> print(solution.y.mean)
+    [[0.15      ]
+     [0.2076198 ]
+     [0.27932997]
+     [0.3649165 ]
+     [0.46054129]
```

Comment:
this doctest has not been touched -- it passes in the same way (so do all the other tests). It is just an internal change, the API is identical to before.